### PR TITLE
merge details with referenced field

### PR DESCRIPTION
**Pull request by MrCrayon**
Before removing the overridden DataRow gets details fields and merges it in the displayed DataRow.
Solves #2182

### CODE REVIEW

**Comment by emptynick** [comment]
I dont really understand what this does.
~~No matter what, #2182 is about adding details to a relationship, not merging them back once deleted.~~

**Comment by MrCrayon** [comment]
@emptynick I should have probably explained it further.
When you create a relationship field you can't choose details because that field is hidden and used to define the relationship, but you can set details for the field associated to that relationship.
As example let's say I have Posts with category_id, I can set details for that field, then I add a relationship belongsTo Category using category_id as foreign key.

When Bread for Posts is shown removeRelationshipField is called and that removes category_id with all information stored in its details field.

My commit gets on the fly the details from category_id and merge it with relationship field details, so that when relationship field is shown it carries the details of the related field.

Basically is the same as the workaround proposed in issue #2182 but it's done on the fly without modifying db data.

BTW I checked that there is no conflict between documented keywords used in details field and keywords used to define relationship.

**Comment by emptynick** [comment]
Ok I understand, but this only works on Belongs To Relationships then? Or does it also take the details from a foreign column?

**Comment by MrCrayon** [comment]
Yes, because for other relationship types there is no field referenced in the same table.
A better solution would be to move relationship data in another field and allow details to be customized like any other field.
This is just a temporary solution, I think all BREAD system is under reconstruction so a more complicated solution might not be needed.

**Comment by emptynick** [comment]
Each relationship has its own `data_row` so we _could_ display a normal JSON-editor, maybe in a `collapsible` so it is hidden first

**Comment by MrCrayon** [comment]
The problem I see is that you will have at the same time two ways to change that data, JSON-editor and current interface.
One way to solve this duplicity is to remove the relationship data before displaying JSON-editor and merge it again before saving to db.
